gprofiler/spark/spark_collector.py

```diff
@@ -0,0 +1,894 @@
+                    starts with the host name: {host_name}, enabling spark collector..."
+            )
+        else:
+            self._logger.info(
+                f"this is not the collector master, because rm1: {rm1_address}\
+                     does not starts with the host name: {host_name}, skipping spark\
+                         collection on this yarn master..."
+            )
+        return is_collection_master
+
+    def _get_spark_manager_process(self) -> Optional[psutil.Process]:
+        try:
+            return next(
+                search_for_process(
+                    lambda process: "org.apache.hadoop.yarn.server.resourcemanager.ResourceManager" in process.cmdline()
+                    or "org.apache.spark.deploy.master.Master" in process.cmdline()
+                    or "mesos-master" in process.exe()
+                )
+            )
+        except StopIteration:
+            return None
+
+    def _find_spark_cluster(self) -> Optional[Tuple[str, str]]:
+        """:return: (master address, cluster mode)"""
+        spark_master_process = self._get_spark_manager_process()
+        spark_cluster_mode = "unknown"
+        webapp_url = None
+
+        if spark_master_process is None:
+            self._logger.debug("Could not find any spark master process (resource manager or spark master)")
+            return None
+
+        if "org.apache.hadoop.yarn.server.resourcemanager.ResourceManager" in spark_master_process.cmdline():
+            if not self._is_yarn_master_collector(spark_master_process):
+                return None
+            spark_cluster_mode = SPARK_YARN_MODE
+            webapp_url = self._guess_yarn_resource_manager_webapp_address(spark_master_process)
+        elif "org.apache.spark.deploy.master.Master" in spark_master_process.cmdline():
+            spark_cluster_mode = SPARK_DRIVER_MODE
+            webapp_url = self._guess_driver_application_master_address(spark_master_process)
+        elif "mesos-master" in spark_master_process.exe():
+            spark_cluster_mode = SPARK_MESOS_MODE
+            webapp_url = self._guess_mesos_master_webapp_address(spark_master_process)
+
+        if spark_master_process is None or webapp_url is None or spark_cluster_mode == "unknown":
+            self._logger.warning("Could not get proper spark cluster configuration")
+            return None
+
+        self._logger.info("Guessed settings are", extra={"cluster_mode": spark_cluster_mode, "webbapp_url": webapp_url})
+
+        return webapp_url, spark_cluster_mode
+
+    def start(self) -> bool:
+        spark_cluster_conf = self._find_spark_cluster()
+        if spark_cluster_conf is None:
+            self._logger.debug("Could not guess spark configuration, probably not master node")
+            return False
+
+        master_address, cluster_mode = spark_cluster_conf
+        self._spark_sampler = SparkCollector(cluster_mode, master_address)
+        self._stop_event.clear()
```

Comment:
This is not required

Reply:
why would it not be required?

Reply:
Well the instance was just created. Anyway, yeah, we can keep